core/templates/base-components/i18n-footer.component.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for changing translation language.
+ */
+
+import { Component } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { AppConstants } from 'app.constants';
+import { I18nLanguageCodeService } from 'services/i18n-language-code.service';
+import { SiteLanguageBackendApiService } from 'services/site-language-backend-api.service';
+import { UserService } from 'services/user.service';
+
+interface SiteLanguage {
+  readonly id: string,
+  readonly text: string,
+  readonly direction: string
+}
+
+@Component({
+  selector: 'oppia-i18n-footer',
+  templateUrl: './i18n-footer.component.html'
+})
+export class I18nFooterComponent {
+  currentLanguageCode: string;
+  supportedSiteLanguages: readonly SiteLanguage[];
```

Comment:
are we expecting this to be `readonly supportedSiteLanguages: SiteLanguage[];`? (not sure!)

Reply:
![Screenshot from 2021-05-30 17-21-29](https://user-images.githubusercontent.com/24855641/120103100-a6db8d80-c16b-11eb-9368-9e83c894ecc1.png)
This is the implicit data type inferred by typescript.
It shows an error if `readonly` is not used.